pyodide_build/tests/test_venv.py

```diff
@@ -0,0 +1,225 @@
+    )
+    monkeypatch.setattr(
+        "pyodide_build.out_of_tree.venv.create_pip_script", lambda venv_bin: None
+    )
+    monkeypatch.setattr(
+        "pyodide_build.out_of_tree.venv.create_pyodide_script", lambda venv_bin: None
+    )
+    monkeypatch.setattr(
+        "pyodide_build.out_of_tree.venv.install_stdlib", lambda venv_bin: None
+    )
+    monkeypatch.setattr(
+        "pyodide_build.out_of_tree.venv.pyodide_dist_dir",
+        lambda: temp_venv_path / "dist",
+    )
+
+    # necessary directories for valid venv
+    (temp_venv_path / "dist").mkdir(exist_ok=True)
+    (temp_venv_path / "bin").mkdir(exist_ok=True)
+
+    venv.create_pyodide_venv(temp_venv_path, options)
+
+    for expected_arg in expected_calls:
+        assert (
+            expected_arg in captured_args
+        ), f"Expected {expected_arg} in call args: {captured_args}"
+
+
+def test_supported_virtualenv_options():
+    """Test that all (currently) supported options are in SUPPORTED_VIRTUALENV_OPTIONS"""
+    supported_options = venv.SUPPORTED_VIRTUALENV_OPTIONS
+    expected_options = [
+        "--clear",
+        "--no-clear",
+        "--no-vcs-ignore",
+        "--symlinks",
+        # "--copies",
+        # "--always-copy",
+        # "--symlink-app-data",
+        "--no-download",
+        "--never-download",
+        "--download",
+        "--extra-search-dir",
+        "--pip",
+        "--setuptools",
+        "--wheel",
+        "--no-setuptools",
+        "--no-wheel",
+        "--no-periodic-update",
+    ]
+
+    assert set(supported_options) == set(expected_options)
+
+
+@pytest.mark.skipif(True, reason="Requiring full Pyodide environment")
+@pytest.mark.parametrize(
+    "options,check_function",
+    [
+        (
+            [],
+            lambda path: (path / "bin" / "python").exists()
+            and (path / "bin" / "pip").exists(),
+        ),
+        (["--clear"], lambda path: (path / "bin" / "python").exists()),
+        (["--no-vcs-ignore"], lambda path: (path / "bin" / "python").exists()),
+        (
+            ["--no-setuptools"],
+            lambda path: not list(path.glob("**/setuptools-*.dist-info")),
+        ),
+        (["--no-wheel"], lambda path: not list(path.glob("**/wheel-*.dist-info"))),
+    ],
+    ids=["default", "clear", "no-vcs-ignore", "no-setuptools", "no-wheel"],
+)
+def test_integration_venv_creation(temp_venv_path, options, check_function):
+    try:
+        venv.create_pyodide_venv(temp_venv_path, options)
+        assert temp_venv_path.exists()
+        assert (temp_venv_path / "bin" / "python").exists()
+        assert (temp_venv_path / "bin" / "pyodide").exists()
+        assert (temp_venv_path / "pip.conf").exists()
+        assert check_function(temp_venv_path)
+
+    except Exception as e:
+        pytest.fail(f"Failed to create virtual environment: {e}")
+
+
+@pytest.mark.skipif(True, reason="Integration test requiring full Pyodide environment")
```

Comment:
These tests are skipped for now, but I'll refactor them later in the day. They are not that slow, but they do take a minute, so I think it makes sense to include them in the integration tests job (one of them also queries PyPI).

Reply:
This is resolved with e32a0cca3e8ebda0e7cfc2e6fc3fb12deb533ae6, the tests are a bit faster now.